signature/mechanism.go

```diff
@@ -4,9 +4,13 @@ package signature
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
+	"io/ioutil"
+	"strings"
 
 	"github.com/mtrmac/gpgme"
+	"golang.org/x/crypto/openpgp"
```

Comment:
This does not add any _new_ uses of `gpgme`. Getting rid of gpgme happens in #207 not here (though the two do touch the same code, so the latter one will have to be modified. I expect the newly added `UntrustedSignatureContents` to remain in `mechanism.go` as a shared helper, called by both GPG backends.)